genai-perf/genai_perf/inputs/retrievers/synthetic_prompt_generator.py

```diff
@@ -31,95 +32,74 @@ def create_synthetic_prompt(
         prompt_tokens_stddev: int = 250,
     ) -> str:
         """
-        Generate a prompt that randomly samples lines from
-        Washington's farewell address at farewell.txt.
+        Generate a synthetic prompt with a specific number of tokens.
 
         Args:
-            prompt_tokens_mean:
-                The mean length of the prompt to generate
-            prompt_tokens_stddev:
-                The standard deviation of the length of the prompt to generate
+            tokenizer: Tokenizer instance.
+            prompt_tokens_mean: Mean number of tokens in the prompt.
+            prompt_tokens_stddev: Standard deviation for the number of tokens in the prompt.
 
         Returns:
-            The prompt.
+            A synthetic prompt as a string.
         """
+        if cls._tokenized_corpus is None:
+            cls._initialize_corpus(tokenizer)
 
-        num_prompt_tokens = SyntheticPromptGenerator._sample_random_positive_int(
-            prompt_tokens_mean, prompt_tokens_stddev
-        )
-
-        farewell_lines = SyntheticPromptGenerator._create_farewell_lines()
-        prompt = SyntheticPromptGenerator._create_prompt_from_lines(
-            num_prompt_tokens, farewell_lines, tokenizer
+        num_prompt_tokens = max(
+            1, int(random.gauss(prompt_tokens_mean, prompt_tokens_stddev))
         )
 
-        return prompt
+        return cls._generate_prompt(tokenizer, num_prompt_tokens)
 
     @classmethod
-    def _create_farewell_lines(cls) -> List[str]:
-        farewell_path = pathlib.Path(__file__).parent.resolve() / "farewell.txt"
+    def _initialize_corpus(cls, tokenizer: Tokenizer):
+        """
+        Load and tokenize the corpus once, storing it for reuse.
+
+        Args:
+            tokenizer: Tokenizer for tokenizing the corpus.
+        """
+        farewell_path = pathlib.Path(__file__).parent / "sonnets.txt"
+
         with open(farewell_path, "r") as f:
-            farewell_lines = f.readlines()
-        random.shuffle(farewell_lines)
+            lines = f.readlines()
+
+        def tokenize_chunk(chunk):
+            return tokenizer.encode(" ".join(chunk))
+
+        num_threads = 150
```

Comment:
Nice catch! I was doing empirical testing in case the I/O made a difference, but this approach works better. And shaved off another ~.06s on my machine. Updated!